onadata/apps/logger/management/commands/transfer_project.py

```diff
@@ -0,0 +1,97 @@
+"""Functionality to transfer a project form one owner to another."""
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from onadata.apps.logger.models import Project, XForm, DataView, MergedXForm
+from onadata.apps.logger.models.project import set_object_permissions \
+    as set_project_permissions
+from onadata.libs.utils.project_utils import set_project_perms_to_xform
+
+
+class Command(BaseCommand):  # pylint: disable=C0111
+    help = 'A command to reassign a project form one user to the other.'
+
+    errors = []
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--currentowner',
+            help='Username of the current owner of of the projects',
+        )
+        parser.add_argument(
+            '--newowner',
+            help='TUsername of new owner of the projects',
+        )
+
+    def get_user(self, username):  # pylint: disable=C0111
+        user_model = get_user_model()
+        user = None
+        try:
+            user = user_model.objects.get(username=username)
+        except user_model.DoesNotExist:
+            self.errors.append("User {0} does not exist \n".format(username))
+        return user
+
+    def update_xform_with_new_user(self, project, user):
+        """
+        Update XForm user update the DataViews and also set the permissions
+        for the xform and the project.
+        """
+        xforms = XForm.objects.filter(project=project)
```

Comment:
We may need to address the issue for inactive forms i.e `downloadable=False`, there is usually an exception that is raised for inactive forms. I am not sure if it applies to the other models. Also, for the situation where an object has been soft deleted.

Reply:
I have made changes to address both scenarios.